lib/Dredd.js

```diff
@@ -40,215 +99,89 @@ class Dredd {
+      }
+
+      callback(null, apiDescriptionsWithTransactions);
+    });
   }
 
   run(callback) {
     this.logger.debug('Resolving --require');
     if (this.configuration.options.require) {
-      let mod = this.configuration.options.require;
-      const abs = fs.existsSync(mod) || fs.existsSync(`${mod}.js`);
-      if (abs) {
-        mod = path.resolve(mod);
-      }
+      const requirePath = resolveModule(this.configuration.custom.cwd, this.configuration.options.require);
       try {
-        // eslint-disable-next-line global-require, import/no-dynamic-require
-        require(mod);
-      } catch (err) {
-        this.logger.error(`Error requiring module '${mod}':`, err.message);
-        callback(err, this.stats);
+        require(requirePath); // eslint-disable-line global-require, import/no-dynamic-require
+      } catch (error) {
+        callback(error, this.stats);
         return;
       }
     }
 
-    this.logger.debug('Resolving API descriptions locations');
-    try {
-      this.files = resolveLocations(this.configuration.custom.cwd, this.configuration.options.path);
-    } catch (err) {
-      callback(err, this.stats);
-      return;
-    }
-
-    // Spin that merry-go-round
-    this.logger.debug('Reading API description files.');
-    this.loadFiles((loadErr) => {
-      if (loadErr) { return callback(loadErr, this.stats); }
-
-      this.logger.debug('Parsing API description files and compiling a list of HTTP transactions to test.');
-      this.compileTransactions((compileErr) => {
-        if (compileErr) { return callback(compileErr, this.stats); }
-
-        this.logger.debug('Starting reporters and waiting until all of them are ready.');
-        this.emitStart((emitStartErr) => {
-          if (emitStartErr) { return callback(emitStartErr, this.stats); }
-
-          this.logger.debug('Starting transaction runner.');
-          this.startRunner((runnerErr) => {
-            if (runnerErr) { return callback(runnerErr, this.stats); }
-
-            this.logger.debug('Wrapping up testing.');
-            this.transactionsComplete(callback);
-          });
-        });
+    this.logger.debug('Configuring reporters');
+    configureReporters(this.configuration, this.stats, this.transactionRunner);
+    // FIXME: 'configureReporters()' pollutes the 'stats' object with
+    // this property. Which is unfortunate, as the 'stats' object is
+    // a part of Dredd's public interface. This line cleans it up for now, but
+    // ideally the property wouldn't be needed at all.
+    delete this.stats.fileBasedReporters;
+
+    this.logger.debug('Preparing API description documents');
+    this.prepareAPIdescriptions((error, apiDescriptions) => {
+      if (error) { callback(error, this.stats); return; }
+      this.configuration.apiDescriptions = apiDescriptions;
+
+      // TODO https://github.com/apiaryio/dredd/issues/1191
+      const annotationsError = handleRuntimeProblems(apiDescriptions, this.logger);
```

Comment:
I want to get rid of `handleRuntimeProblems()` as the next step after this PR, finally unblocking upgrading Dredd Transactions.